dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EventCategoryOptValidationHook.java

```diff
@@ -0,0 +1,226 @@
+                .getTrackedEntityInstance( bundle, event.getTrackedEntity() );
+            Program program = PreheatHelper.getProgram( bundle, event.getProgram() );
+
+            if ( program == null )
+            {
+                continue;
+            }
+
+            programStage = (programStage == null && program.isWithoutRegistration())
+                ? program.getProgramStageByStage( 1 ) : programStage;
+            if ( programStage == null )
+            {
+                continue;
+            }
+
+            programInstance = getProgramInstance( actingUser, programInstance, trackedEntityInstance, program );
+            program = programInstance.getProgram();
+
+            validateCategoryOptionCombo( bundle, reporter, actingUser, event, program );
+        }
+
+        return reporter.getReportList();
+    }
+
+    private void validateCategoryOptionCombo( TrackerBundle bundle, ValidationErrorReporter errorReporter
+        , User user, Event event, Program program )
+    {
+
+        // NOTE: Morten H. & Stian. Abyot : How do we solve this in the new importer?
+//        CategoryOptionCombo categoryOptionCombo;
+//        if ( (event.getAttributeCategoryOptions() != null
+//            && program.getCategoryCombo() != null)
+//            || event.getAttributeOptionCombo() != null )
+//        {
+//            try
+//            {
+//                categoryOptionCombo = getAttributeOptionCombo( bundle,
+//                    program.getCategoryCombo(),
+//                    event.getAttributeCategoryOptions(),
+//                    event.getAttributeOptionCombo() );
+//            }
+//            catch ( IllegalQueryException e )
+//            {
+//                errorReporter.addError( newReport( TrackerErrorCode.E1072 )
+//                    .addArg( event.getAttributeCategoryOptions() )
+//                    .addArg( e.getMessage() ) );
+//                return;
+//            }
+//        }
+//        else
+//        {
+//            categoryOptionCombo = (CategoryOptionCombo) bundle.getPreheat().getDefaults()
+//                .get( CategoryOptionCombo.class );
+//        }
+
+        CategoryOptionCombo categoryOptionCombo = (CategoryOptionCombo) bundle.getPreheat().getDefaults()
+            .get( CategoryOptionCombo.class );
+        if ( categoryOptionCombo == null )
+        {
+            errorReporter.addError( newReport( TrackerErrorCode.E1055 ) );
+            return;
+        }
+
+        if ( categoryOptionCombo.isDefault()
+            && program.getCategoryCombo() != null
+            && !program.getCategoryCombo().isDefault() )
+        {
+            errorReporter.addError( newReport( TrackerErrorCode.E1055 ) );
+            return;
+        }
+
+        // NOTE: How to best get current date into iso format?
+        Date eventDate = DateUtils.parseDate( ObjectUtils
+            .firstNonNull( event.getEventDate(), event.getDueDate(), DateUtils.getIso8601( new Date() ) ) );
```

Comment:
Assigning today for eventDate does not look right. 

If event comes without both event and schedule date, then we should reject.